### PR TITLE
Swap nodejs-prybar some bugs for others

### PR DESCRIPTION
**Pull request by masad-frost**
TL;DR nodejs-prybar implementation now matches https://github.com/replit/goval/blob/master/images/nodejs/interp.js including the bugs so we can migrate nodejs to prybar without changing any behavior.

We want to start using prybar node on replit, the prybar implementation has a bug where defining a variable in the main file leads to it defined being defined in the global scope. This bug can be seen in the new test `main_module_scoped_to_global` which now passes.

I changed the implementation to use a custom nodejs context instead of using a global context which seems to solve this problem. However, it introduces a new bug where `global`s in the main file are not the same as globals in submodules :eyes:. This is because we inject `require` into the context from the global scope meaning submodule's are actually running in the `repl.js` context :zany_face:, we'd have to implement `require` and even possibly the whole `module` in userspace (or actually pry node open) to get around this. Some inspiration https://github.com/patriksimek/vm2 and https://github.com/felixge/node-sandboxed-module/blob/master/lib/sandboxed_module.js

The new bug can be seen in `global_bug` test which passed before and now fails (i commented out the part that makes it fail).

### CODE REVIEW

**Comment by masad-frost** [comment]
We should also look into vm.Module at some point.

**Comment by masad-frost** [comment]
@AllAwesome497 suggested something like this  https://replit.com/@AllAwesome497/InfantileVagueDevelopers#index.js. Which is similar to `node -r index.js` but allows us to modify the ps1 and other things.

The idea is that we require the user's module with a footer function that adds a repl to it using a require hook.

Let's get this PR merged and shipped, use prybar-node, then we can iterate.

**Review by cbrewster — APPROVED** [review]
:lady_beetle:
